### PR TITLE
feat(FEC-9545): add ability to correlate udrm calls to player type and session

### PR DESCRIPTION
**Pull request by RoyBregman**
### Description of the Changes

In addKalturaParams, if a source has udr licenseurl then append sessionid, clienttag, referrer and uiConfId as query params.

### CheckLists

- [X] changes have been done against master branch, and PR does not conflict
- [X] new unit / functional tests have been added (whenever applicable)
- [X] test are passing in local environment
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: src/common/utils/kaltura-params.js [hunk 1]
```diff
@@ -59,23 +64,26 @@ function setSessionId(playerConfig: PartialKPOptionsObject, sessionId: string):
...
+ * @param {string} url - url
  * @param {string} sessionId - session id
- * @return {void}
+ * @param {string} paramName - optional param name of the session id
+ * @return {string} - the url with the new sessionId
  * @private
  */
-function updateSessionIdInUrl(source: Object = {}, sessionId: ?string): void {
+function updateSessionIdInUrl(url: string, sessionId: ?string, paramName: ?string): string {
+  paramName = paramName || PLAY_SESSION_ID;
```
Comment (OrenMe) [inline src/common/utils/kaltura-params.js]: either make it mandatory or not, doesn't make a lot of sense to make it optional

File: src/common/utils/kaltura-params.js [hunk 2]
```diff
@@ -94,28 +102,45 @@ function getReferrer(): string {
...
+ * @return {string} - the url with the client tag appended in the query params
  * @private
  */
-function addClientTag(source: PKMediaSourceObject): void {
-  if (source.url.indexOf(CLIENT_TAG) === -1) {
-    let delimiter = source.url.indexOf('?') === -1 ? '?' : '&';
-    source.url += delimiter + CLIENT_TAG + __VERSION__;
+function addClientTag(url: string): string {
+  if (url.indexOf(CLIENT_TAG) === -1) {
+    let delimiter = url.indexOf('?') === -1 ? '?' : '&';
```
Comment (OrenMe) [inline src/common/utils/kaltura-params.js]: I see we have this a lot so better to make this a local method and use in all places

File: src/common/utils/kaltura-params.js [hunk 3]
```diff
@@ -133,13 +158,28 @@ function addKalturaParams(player: Player, playerConfig: PartialKPOptionsObject):
...
-          addClientTag(source);
+          source.url = updateSessionIdInUrl(source.url, playerConfig.session && playerConfig.session.id);
+          source.url = addReferrer(source.url);
+          source.url = addClientTag(source.url);
+        }
+        if (source.drmData && source.drmData.length) {
+          source.drmData.forEach(drmData => {
+            if (
+              typeof drmData.licenseUrl === 'string' &&
+              drmData.licenseUrl.indexOf(UDRM_DOMAIN) > -1 &&
```
Comment (OrenMe) [inline src/common/utils/kaltura-params.js]: a bit of sugar syntax but maybe do:
```javascript
[UDRM_DOMAIN, CUSTOM_DATA, SIGNATURE].every(t => drmData.licenseUrl.includes(t));
```

File: src/common/utils/kaltura-params.js [hunk 4]
```diff
@@ -94,28 +102,45 @@ function getReferrer(): string {
...
+ * @param {string} url - url
+ * @param {PartialKPOptionsObject} playerConfig - player config
+ * @return {string} - the url with the uiconf id appended in the query params
+ * @private
+ */
+function addUIConfId(url: string, playerConfig: PartialKPOptionsObject): string {
+  if (url.indexOf(UICONF_ID) === -1 && playerConfig.session && typeof playerConfig.session.uiConfId === 'number') {
+    let delimiter = url.indexOf('?') === -1 ? '?' : '&';
+    // $FlowFixMe
+    url += delimiter + UICONF_ID + playerConfig.session.uiConfId;
```
Comment (Yuvalke) [inline src/common/utils/kaltura-params.js]: The uiConfId comes from playerConfig.provider, playerConfig.session set by provider.
so check if it's early enough

File: src/common/utils/kaltura-params.js [hunk 5]
```diff
@@ -133,13 +158,28 @@ function addKalturaParams(player: Player, playerConfig: PartialKPOptionsObject):
...
+        }
+        if (source.drmData && source.drmData.length) {
+          source.drmData.forEach(drmData => {
+            if (
+              typeof drmData.licenseUrl === 'string' &&
+              drmData.licenseUrl.indexOf(UDRM_DOMAIN) > -1 &&
+              drmData.licenseUrl.indexOf(CUSTOM_DATA) > -1 &&
+              drmData.licenseUrl.indexOf(SIGNATURE) > -1
+            ) {
+              drmData.licenseUrl = updateSessionIdInUrl(drmData.licenseUrl, playerConfig.session && playerConfig.session.id, DRM_SESSION_ID);
```
Comment (OrenMe) [inline src/common/utils/kaltura-params.js]: we use `playerConfig.session && playerConfig.session.id` twice now so take it outside and above(line `sources[key].forEach(source => {`) and then reference it in both the source and drm urls call to `updateSessionIdInUrl`

File: src/common/utils/kaltura-params.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (RoyBregman) [reply src/common/utils/kaltura-params.js]: was a problem with flow - didn't identify the if correctly and said it was possible to be undefined. Fixed

File: src/common/utils/kaltura-params.js (hunk 1 above)
Comment (RoyBregman) [reply src/common/utils/kaltura-params.js]: fixed - please see i understood you correctly

File: src/common/utils/kaltura-params.js (hunk 2 above)
Comment (RoyBregman) [reply src/common/utils/kaltura-params.js]: done

File: src/common/utils/kaltura-params.js (hunk 3 above)
Comment (RoyBregman) [reply src/common/utils/kaltura-params.js]: done

File: src/common/utils/kaltura-params.js (hunk 5 above)
Comment (RoyBregman) [reply src/common/utils/kaltura-params.js]: done

File: src/common/utils/kaltura-params.js (hunk 4 above)
Comment (RoyBregman) [reply src/common/utils/kaltura-params.js]: @OrenMe , uiConfId can come from provider and from session. 
Maybe I should make a similar check like here:
https://github.com/kaltura/kaltura-player-js/blob/master/src/common/plugins/plugins-config.js#L59 ?

Comment (OrenMe) [reply src/common/utils/kaltura-params.js]: Agree, check from provider config

Comment (RoyBregman) [reply src/common/utils/kaltura-params.js]: changed to provider

**Pull request merged**